src/components/Microsoft.DotNet.UpgradeAssistant.Extensions/ExtensionProviderExtensions.cs

```diff
@@ -96,6 +97,27 @@ public static OptionsBuilder<ExtensionOptions> AddDefaultExtensions(this Options
             });
         }
 
+        public static OptionsBuilder<ExtensionOptions> AddExtensionOption<TOption>(this OptionsBuilder<ExtensionOptions> builder, TOption option)
+        {
+            if (builder is null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            var json = JsonSerializer.SerializeToUtf8Bytes(option);
+            using var stream = new MemoryStream(json);
+
+            var config = new ConfigurationBuilder()
```

Comment:
Should we also include configuration from appsettings.json since that's a logical place people might put default extension settings at some point?

Reply:
I don't think so - they should be putting anything like that into the ExtensionManfiest.json. This is to pass through command line args.

Reply:
Ok, that makes sense. I'm still in the mindset of when we had a default extension mixed in with the tool (and appsettings.json served as the extensionmanifest.json). You're right that now the extensions all have their own manifests that should be used.